### ISSUE TITLE
youtube 2019上的测试结果

### ISSUE THREAD

**Issue by karchen-geng** [hkchengrex/STCN#142]
您好，我用您repo提供的stcn_s03权重文件，得到如下的精度，与您文章中的82.7 相差几个百分点，请问这样的情况正常吗？
![图片](https://github.com/hkchengrex/STCN/assets/81698409/579adb54-ad92-4e84-999d-c1e7e8fb583a)

**Comment by hkchengrex** [comment]
It's not normal. The result should be exactly the same as our provided pre-computed masks.

**Comment by karchen-geng** [comment]
权重是stcn_s03，测试代码为eval_youtube,分割的结果直接提交CodaLab中，为什么结果相差这么多，您有什么好的建议嘛》？

**Comment by hkchengrex** [comment]
Can you check if:
1. Submitting our pre-computed results gives the expected performance
2. The library versions match
3. Your computed results are the same as our pre-computed ones

**Comment by karchen-geng** [comment]
请问我如何得到您的pre-computed results，是通过您给的权重通过eval_youtube得到嘛？，还有您用的youtube2019是valid，还是valid_all_frames？

**Comment by hkchengrex** [comment]
Yes. It should use all_frames by default.

**Comment by karchen-geng** [comment]
您好，我注意到一个问题：之前的环境版本低，所以变换用的这个Image库，但原文用的是InterpolationMode库，请问是不是这个也会影响性能？
![图片](https://github.com/hkchengrex/STCN/assets/81698409/955ca11f-fe37-48a7-b2e4-5a93102151d0)

![图片](https://github.com/hkchengrex/STCN/assets/81698409/d79946f1-08af-4e0c-af02-2b5f46c0ecd8)
![图片](https://github.com/hkchengrex/STCN/assets/81698409/28f12b79-9beb-4462-a442-1219c2eee77b)

还有相同的权重文件针对valid_all_frames 和valid得到的结果， 送到codaLab中测试时，两个是否会也有性能差异，这个您知道嘛？

**Comment by hkchengrex** [comment]
1. They are the same.
2. We always use all_frames. In STCN, if you use valid, you can set mem_every=1 for the same output. This is not true in general.

**Comment by karchen-geng** [comment]
那请问您训练的时候的train还是train- all- frames？在 2023年5月18日，11:13，Rex Cheng ***@***.***> 写道：﻿

They are the same.
We always use all_frames. In STCN, if you use valid, you can set mem_every=1 for the same output. This is not true in general.

**Comment by hkchengrex** [comment]
You cannot train on all_frames.

**Comment by karchen-geng** [comment]
非常感谢，我测试一下修改mem_eve之后的效果。

**Comment by karchen-geng** [comment]
您好，我将memory修改为1后效果为81.7，非常感谢您的帮助。请问youtube18如何进行测试呢？18年的测试平台已经更新，请问您有评估的方法嘛？
![图片](https://github.com/hkchengrex/STCN/assets/81698409/c8f59b69-e8b3-4d5a-95bf-9a88fc7f0d81)

**Comment by hkchengrex** [comment]
https://youtube-vos.org/dataset/vos/

**Comment by karchen-geng** [comment]
非常感谢！

**Comment by karchen-geng** [comment]
您好，我想问一下假设s0阶段的权重有（iter：250k和iter：300k两个），iter：250k在davis2017的分割结果要比iter：300k的结果好一点，但是iter：300k是最后的那个权重，请问我用这两个权重中的哪一个，进行stage3的训练会比较好，您有什么建议嘛？

**Comment by karchen-geng** [comment]
还有一个问题，stage2中我的iteration变了，steps是不是也应该按比例改变？

**Comment by hkchengrex** [comment]
> 您好，我想问一下假设s0阶段的权重有（iter：250k和iter：300k两个），iter：250k在davis2017的分割结果要比iter：300k的结果好一点，但是iter：300k是最后的那个权重，请问我用这两个权重中的哪一个，进行stage3的训练会比较好，您有什么建议嘛？

I haven't tried using weights other than the last one. I don't think there will be any major differences.



> 您好，原文stage3中您的单GPU-batchsize为4，iter为150k（用到两个GPU），我的实验中共一个GPU，故我将batch设置为8，我的iter是否也应该为150k？学习率是否需要调动呢？

Keep the number of iterations and the learning rate the same.

**Comment by karchen-geng** [comment]
感谢！！！
